src/Param.cc

```diff
@@ -427,6 +435,106 @@ bool ParseColorUsingStringStream(const std::string &_input,
   return isValidColor;
 }
 
+//////////////////////////////////////////////////
+/// \brief Helper function for Param::ValueFromString for parsing pose. This
+/// checks the pose components specified in _input are xyzrpy
+/// (expects 6 values) and whether to parse the rotation values as degrees using
+/// the parent element attributes.
+/// \param[in] _input Input string.
+/// \param[in] _key Key of the parameter, used for error message.
+/// \param[in] _attributes Attributes associated to this pose.
+/// \param[out] _value This will be set with the parsed value.
+/// \return True if parsing pose succeeded.
+bool ParsePoseUsingStringStream(const std::string &_input,
+    const std::string &_key, const Param_V &_attributes,
+    ParamPrivate::ParamVariant &_value)
+{
+  StringStreamClassicLocale ss(_input);
+  std::string token;
+  std::array<double, 6> values;
+  std::size_t valueIndex = 0;
+  double v;
+  bool isValidPose = true;
+  while (ss >> token)
+  {
+    try
+    {
+      v = std::stod(token);
+    }
+    // Catch invalid argument exception from std::stod
+    catch(std::invalid_argument &)
+    {
+      sdferr << "Invalid argument. Unable to set value ["<< _input
+             << "] for key [" << _key << "].\n";
+      isValidPose = false;
+      break;
+    }
+    // Catch out of range exception from std::stod
+    catch(std::out_of_range &)
+    {
+      sdferr << "Out of range. Unable to set value [" << token
+             << "] for key [" << _key << "].\n";
+      isValidPose = false;
+      break;
+    }
+
+    if (!std::isfinite(v))
+    {
+      sdferr << "Pose values must be finite.\n";
+      isValidPose = false;
+      break;
+    }
+
+    if (valueIndex >= 6u)
+    {
+      sdferr << "Pose values can only accept 6 values.\n";
+      isValidPose = false;
+      break;
+    }
+
+    values[valueIndex++] = v;
+  }
+
+  if (!isValidPose)
+    return false;
+
+  if (values.size() != 6u)
+  {
+    sdferr << "The value for //pose must have 6 values, but "
+        << values.size() << " were found instead.\n";
+    return false;
+  }
```

Comment:
Now that we're using arrays, this will always be false. 

```suggestion
  if (valuesIndex != 6u)
  {
    sdferr << "The value for //pose must have 6 values, but "
        << valuesIndex << " were found instead.\n";
    return false;
  }
```

Reply:
:facepalm: Thanks for catching that! Fixed in https://github.com/ignitionrobotics/sdformat/pull/589/commits/95c292e922b6d94f321660d47e1debdd11a1b573